client/rest-high-level/src/main/java/org/elasticsearch/client/security/GetSslCertificatesResponse.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.client.security;
+
+import org.elasticsearch.client.security.support.CertificateInfo;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.xcontent.DeprecationHandler;
+import org.elasticsearch.common.xcontent.NamedXContentRegistry;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentFactory;
+import org.elasticsearch.common.xcontent.XContentParser;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Response object when retrieving the X.509 certificates that are used to encrypt communications in an Elasticsearch cluster.
+ * Returns a list of {@link CertificateInfo} objects describing each of the certificates.
+ */
+public final class GetSslCertificatesResponse {
+
+    private final List<CertificateInfo> certificates;
+
+    public GetSslCertificatesResponse(List<CertificateInfo> certificates) {
+        this.certificates = certificates;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        final GetSslCertificatesResponse that = (GetSslCertificatesResponse) o;
+        return Objects.equals(this.certificates, that.certificates);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(certificates);
+    }
+
+    public static GetSslCertificatesResponse fromXContent(XContentParser parser) throws IOException {
+        List<Object> unparsedCerts = parser.list();
+        if (unparsedCerts.isEmpty()) {
+            return new GetSslCertificatesResponse(Collections.emptyList());
+        }
+        List<CertificateInfo> certificates = new ArrayList<>();
+        for (Object cert : unparsedCerts) {
+            try (XContentBuilder builder = XContentFactory.jsonBuilder()) {
```

Comment:
I am confused, why can't we just parse using the passed in parser?

Reply:
Because `ConstructingObjectParser` expects a parser with Objects in its `parse()` method but we get an list of objects in the parser. So I had to get the list from the passed in parser and parse each one.
Does this make sense?

Reply:
But `ConstructingObjectParser` can just parse a single node within the parser stream, so you can do something like this _(untested, minimal error checking)_:
```
XContentParserUtils.ensureExpectedToken( Token.START_ARRAY, parser.currentToken(), parser::getTokenLocation);
while (parser.nextToken() != Token.END_ARRAY) {
   certificates.add( CertificateInfo.PARSER.parse(certificateInfoParser, null) );
}
```

Reply:
What Tim said is what I had in mind

Reply:
Gotcha, thanks! I didn't know I could do that. Now that you mentioned it, I saw this being used elsewhere and it makes total sense too.